crates/router/src/routes/routing.rs

```diff
@@ -41,11 +42,17 @@ pub async fn routing_create_config(
         #[cfg(not(feature = "release"))]
         auth::auth_type(
             &auth::HeaderAuth(auth::ApiKeyAuth),
-            &auth::JWTAuth(Permission::RoutingWrite),
+            &auth::JWTAuthProfileFromRoute {
+                profile_id: json_payload.profile_id,
```

Comment:
Let's not use the profile_id from the json in this auth. We can use it if the profile_id is in query param or the url.


If the profile_id is optional and is Noen, we can fallback to normal JWTAuth instead of JWTAuthProfileFromRoute.

Reply:
>If the profile_id is optional and is Noen, we can fallback to normal JWTAuth instead of JWTAuthProfileFromRoute.

Are you suggesting for the places where the profile_id is Optional in query or request params, we will proceed with the validation in core or no validation at all? 
As you are asking to replace the new authProfile functions with older ones.

Reply:
if authentication parameter is coming in request body (profile id in this case) then that check should be done in core function.
if it comes from path then validation should be done before invoking core function.